diff --git a/js/src/jsapi.cpp b/js/src/jsapi.cpp
--- a/js/src/jsapi.cpp
+++ b/js/src/jsapi.cpp
@@ -233,6 +233,17 @@
   if (x < y); // after: bugprone-suspicious-semicolon; readability-braces-around-statements
   {
     x++;
+    struct B {
+      B() {} // after: modernize-use-equals-default
+      B(const B&) {} // after: modernize-use-equals-default
+       B(B &&) {} // after: performance-noexcept-move-constructor
+    };
+
+    struct D : B {
+      D() : B() {}
+      D(const D &RHS) : B(RHS) {} // after: modernize-use-equals-default
+      D(D &&RHS) : B(RHS) {} // after: performance-move-constructor-init; performance-noexcept-move-constructor
+    };
   }
   MOZ_ASSERT(code_ != Uninitialized);
   MOZ_ASSERT(!ok());
